src/platform/src/main/java/org/geoserver/platform/resource/FileSystemWatcher.java

```diff
@@ -35,41 +49,41 @@
  */
 public class FileSystemWatcher implements ResourceNotificationDispatcher, DisposableBean {
 
-    interface FileExtractor {
-        public File getFile(String path);
-    }
+    private static final Logger LOGGER = Logging.getLogger(FileSystemWatcher.class);
 
     /** Change to file system */
     static class Delta {
         final File context;
 
         final Kind kind;
 
-        final List<File> created;
+        final List<String> created;
```

Comment:
What is the reason to move from File to Strings (assume the strings are paths).

Reply:
Because it more closely represents its intent (context directory + relative resource paths), which is exactly what the end-user of this data expects (`ResourceNotification`). Otherwise `ResourceNotification` needs to convert the created/removed/modified files back to relative paths: https://github.com/geoserver/geoserver/pull/3860/files#diff-4edea39a9ddbb6e029dc5a4a6974b7bfL133
Which is also quite some processing and temporary objects we can avoid as we have that information handy at the time of creating the `Delta`: https://github.com/geoserver/geoserver/pull/3860/files#diff-00129441b1b220c30473e61ec5339cc2R245